torchao/dtypes/nf4tensor.py

```diff
@@ -37,11 +41,244 @@ def decorator(func):
     return decorator
 
 
-@implements([torch.ops.aten.detach.default, torch.ops.aten.detach])
+@implements([torch.ops.aten.detach])
 def noop_detach(func, *args, **kwargs):
     return args[0][0]
 
 
+@implements(
```

Comment:
more of a n00b q to @drisspg : what's up with all the `args[0]` I feel like there's some sort of contract I can't quite parse

EDIT: It's the NF4 tensor, could we add some comment somewhere to make this clearer?

Reply:
yeah pretty much args is a tuple of arguments so this is saying take the first one, which is an NF4Tensor. You could honestly try type hinting the `unary` ops with NF4Tensor would probs make the resoning alittle earlier. I try to quickly name the args[0] in the impl since I find that easier to reason about

Reply:
I updated PR with `nf4tensor = args[0]` at the begining to make it clearer